src/snowflake/snowpark/lineage.py

```diff
@@ -527,16 +539,18 @@ def _get_result_dataframe(
         )
         return self._session.create_dataframe(transformed_results, schema=schema)
 
-    def _check_valid_object_name(self, object_name: str) -> None:
+    def _check_valid_object_name(self, object_name: str, object_domain: str) -> None:
         """
         Checks if the object name is one of the below allowed format
-            Non-Case-sensitive: "database.schema.object"
-            Case-sensitive: "\"database\".\"schema\".\"object\""
+            Non-Case-sensitive: "database.schema.object" or "database.schema.object.column_name"
+            Case-sensitive: "\"database\".\"schema\".\"object\"" or "\"database\".\"schema\".\"object\"".\"column_name\"
         """
         parts = _DGQLQueryBuilder.split_fully_qualified_name(object_name)
 
-        # Check if the object name has three parts separated by dots
-        if len(parts) != 3:
+        is_column_domain = object_domain.upper() == _SnowflakeDomain.COLUMN
+        if (is_column_domain and len(parts) != 4) or (
```

Comment:
I didn't know that this API checks that it has to be fully qualified names, can't the caller just pass in partial names like 't1.c1' while the rest can be inferred from the current context? at least DGQL can support that

Reply:
We can make the optimization as a follow up.

Reply:
sg, not blocking on this